packages/cloudflare/src/api/r2-incremental-cache.ts

```diff
@@ -0,0 +1,80 @@
+import { debug, error } from "@opennextjs/aws/adapters/logger.js";
+import type { CacheValue, IncrementalCache, WithLastModified } from "@opennextjs/aws/types/overrides.js";
+import { IgnorableError } from "@opennextjs/aws/utils/error.js";
+
+import { getCloudflareContext } from "./cloudflare-context.js";
+
+/**
+ * An instance of the Incremental Cache that uses an R2 bucket (`NEXT_CACHE_R2_BUCKET`) as it's
+ * underlying data store.
+ *
+ * The directory that the cache entries are stored in can be confused with the `NEXT_CACHE_R2_PREFIX`
+ * environment variable, and defaults to `incremental-cache`.
+ *
+ * The cache uses an instance of the Cache API (`incremental-cache`) to store a local version of the
+ * R2 cache entry to enable fast retrieval, with the cache being updated from R2 in the background.
+ */
+class R2IncrementalCache implements IncrementalCache {
+  readonly name = "r2-incremental-cache";
+
+  async get<IsFetch extends boolean = false>(
+    key: string,
+    isFetch?: IsFetch
+  ): Promise<WithLastModified<CacheValue<IsFetch>> | null> {
+    const r2 = getCloudflareContext().env.NEXT_CACHE_R2_BUCKET;
+    if (!r2) throw new IgnorableError("No R2 bucket");
+
+    debug(`Get ${key}`);
+
+    try {
+      const r2Object = await r2.get(this.getR2Key(key, isFetch));
+      if (!r2Object) return null;
+
+      return {
+        value: await r2Object.json(),
+        lastModified: r2Object.uploaded.getTime(),
+      };
+    } catch (e) {
+      error("Failed to get from cache", e);
+      return null;
+    }
+  }
+
+  async set<IsFetch extends boolean = false>(
+    key: string,
+    value: CacheValue<IsFetch>,
+    isFetch?: IsFetch
+  ): Promise<void> {
+    const r2 = getCloudflareContext().env.NEXT_CACHE_R2_BUCKET;
+    if (!r2) throw new IgnorableError("No R2 bucket");
+
+    debug(`Set ${key}`);
+
+    try {
+      await r2.put(this.getR2Key(key, isFetch), JSON.stringify(value));
+    } catch (e) {
+      error("Failed to set to cache", e);
+    }
+  }
+
+  async delete(key: string): Promise<void> {
+    const r2 = getCloudflareContext().env.NEXT_CACHE_R2_BUCKET;
+    if (!r2) throw new IgnorableError("No R2 bucket");
+
+    debug(`Delete ${key}`);
+
+    try {
+      await r2.delete(this.getR2Key(key));
+    } catch (e) {
+      error("Failed to delete from cache", e);
+    }
+  }
+
+  protected getR2Key(key: string, isFetch?: boolean): string {
+    const directory = getCloudflareContext().env.NEXT_CACHE_R2_PREFIX ?? "incremental-cache";
```

Comment:
nit: could we init `this.directory` in the ctor?

Reply:
it breaks the build as the constructor is called when compiling the opennext config
![image](https://github.com/user-attachments/assets/f9fb34a2-0fdc-4a6e-a907-904635b9c682)